### ISSUE TITLE
Is there a convenient way to switch lantern on/off, or without the lantern page poping out every time?

### ISSUE THREAD

**Issue by hediyi** [getlantern/lantern#4087]
I google all the time and I have to rely on lantern. The problem is I have to quit lantern many times in a day, for several reasons:
- to connect to the the wifi of a local cafe (they require you to type in some pin code on a web page which is not available when lantern is on)
- to watch videos from a domestic website
- to gain faster access to some domestic website

So I repeat this like a dozen times every day: `quit lantern -> reopen it -> close the useless lantern page or switch back to tab I want to browse`. Frankly, this is frustrating sometimes.

What really frustrates me is not that I have to quit and reopen lantern, coz I know that's inevitable. It's there isn't a quick way to do it. So what I'm asking for here is:
- a convenient way to switch lantern on/off, ever better if I don't have to quit lantern
- or at least, provide a setting option to stop the lantern page from popping out every time

Really, Lantern is awesome. That's the only thing I want from lantern.

**Comment by myleshorton** [comment]
Hey @Vince. This is great feedback, thanks! We'd love to make Lantern perfect for you. A couple of questions:

> to connect to the the wifi of a local cafe (they require you to type in some pin code on a web page which is not available when lantern is on)

What's the web page of the wifi? Is it one of those popup windows that sometimes appear? Any idea what address or IP it's loading from?

> to watch videos from a domestic website

Can you give an example of a site? **Do you have Lantern set to "proxy all"?** If not, it should load that page directly already, so you should be able to watch videos just as easily with Lantern running as with Lantern not running.

> to gain faster access to some domestic website

As with the previous, unblocked domestic sites should be just as fast with Lantern running as with it not running.

@oxtoacart this would be an interesting use case for the extension -- i.e. to show when Lantern is proxying a page and when it's not. Would that be useful @esdoppio? If Lantern ran a simple browser extension that made it clear when it's proxying a page and when it's not?

An upcoming version of Lantern does actually have an on/off switch. You can also effectively do the same thing by turning "Manage system proxy" on and off in the settings in 2.1.2.

Ideally, though, we'd love to solve this for you the "right" way and have Lantern just work in all of these cases.

**Comment by louy2** [comment]
> What's the web page of the wifi? Is it one of those popup windows that sometimes appear? Any idea what address or IP it's loading from?

Yes, exactly. It is not available through Lantern because they are mostly served on a LAN server, and for some reason all IP requests go though Lantern regardless whether it's blocked.

There are already proxy switching extensions for both Chrome and Firefox so advanced users can setup so that the extension reads from the PAC served by Lantern. OTOH a no-config extension would definitely offer clearer info for normal users.

**Comment by esdoppio** [comment]
Really appreciate the new switch feature! Does it come with a keyboard shortcut : p @myleshorton 

> What's the web page of the wifi? Is it one of those popup windows that sometimes appear? Any idea what address or IP it's loading from?

Exactly. When lantern is on, it gives a msg of something like "_Still unable to dial tcp://captive.apple.com:80 after 3 attempts_" in the pop-up window. The complete address it's loading from is

http://wiportal.wiwide.com/portal?res=notyet&uamip=10.254.0.1&uamport=3990&challenge=cc6fa101526b2500894e98e0d988709b&called=00-1F-7A-A3-DA-C1&remain=-1&mac=98-E0-D9-88-70-9B&ip=10.254.1.183&ssid=Starbucks&nasid=1&vlan=11&sessionid=572003300000001a&userurl=http%3a%2f%2fwww.ted.com%2f

But I have no problem accessing the page if I put it in a browser even when lantern is on.

> Can you give an example of a site? Do you have Lantern set to "proxy all"?

No, I use the default setting, so **"proxy all" is always off**. Two example sites,

http://v.ifeng.com/
http://tv.sohu.com/

Actually, I'm not 100% sure the problem is caused by using lantern coz I use it 24/7 and only turn it off when a problem occurs. But there were a couple of times when I tried to play a video on a domestic site, I saw some funny msg (in Chinese) like, "_The content is not available in your area/country._" So my first instinct is to shut down lantern and refresh the page. The problem went away. So I just assumed it had something to do with using lantern.

I understand lantern is designed not to affect the access to non-blocked content. But to my experience, it's not quite the case:

- more bad gateway errors occur (refresh a couple of time, good)
- some videos loaded incredibly slow (refresh a couple of times, still, quit lantern, refresh, good)

Those can't be used as evidences. I'm merely stating my experience. These don't happen all the time so I can't reproduce the problem by providing a specific link. And I think it's within acceptable range.

> If Lantern ran a simple browser extension that made it clear when it's proxying a page and when it's not?

Yea I think it's a bonus. This happened to me a couple of times. When I shared some link with friends they told me it's blocked coz I have lantern on all the time I have no idea when a site is blocked or is using content from blocked IPs.

But if I get to choose, I'll stick with how lantern is like now. What I like the most about lantern is when things are going well, it's completely invisible to me. For me, to make lantern perfect is to make it invisible all the time—no UI, never need to switch it on/off, work silently—even make the users forget they live in a censored world. I mean if that's possible : p

**Comment by myleshorton** [comment]
This is very useful @esdoppio and @louy2 thanks. You'd be interested in this too @oxtoacart and @fffw.

The LAN issue is ticketed at #4104

> But there were a couple of times when I tried to play a video on a domestic site, I saw some funny msg (in Chinese) like, "The content is not available in your area/country." So my first instinct is to shut down lantern and refresh the page. The problem went away. So I just assumed it had something to do with using lantern.

This is very interesting. I wonder @fffw if this could be because the direct connection times out too soon, and Lantern proxies it. Would the new version of detour get around that?

> But if I get to choose, I'll stick with how lantern is like now. What I like the most about lantern is when things are going well, it's completely invisible to me. For me, to make lantern perfect is to make it invisible all the time—no UI, never need to switch it on/off, work silently—even make the users forget they live in a censored world. I mean if that's possible : p

That's exactly what we're striving for! The one problem right now is that sometimes when things go wrong we don't know why, so we're likely going to add an extension soon that will help give us more visibility into problems when they occur. So basically the extension will help us achieve exactly what you describe!

**Comment by myleshorton** [comment]
> There are already proxy switching extensions for both Chrome and Firefox so advanced users can setup so that the extension reads from the PAC served by Lantern. OTOH a no-config extension would definitely offer clearer info for normal users.

We've looked at SwitchyOmega and SwitchySharp. Are there others you use @louy2?

**Comment by louy2** [comment]
No.

**Comment by fffw** [comment]
> But there were a couple of times when I tried to play a video on a domestic site, I saw some funny msg (in Chinese) like, "The content is not available in your area/country." So my first instinct is to shut down lantern and refresh the page. The problem went away. So I just assumed it had something to do with using lantern.

@myleshorton It's exactly what new detour want to fix. It won't solve all cases, though. I've seen a few sites respond a simple error page if the client IP is not allowed but take seconds to render a complex page for direct requests. Detour will be spoofed in that case. A quick on/off button still makes sense.
